### ISSUE TITLE
New Rule: no-space-in-emphasis

### ISSUE THREAD

**Issue by Pixel998** [eslint/markdown#378]
### Rule details

This rule is triggered when emphasis markers (bold, italic) are used, but they have spaces between the markers and the text.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
Here is some ** bold ** text.

Here is some * italic * text.

Here is some more __ bold __ text.

Here is some more _ italic _ text.
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Rationale: Emphasis is only parsed as such when the asterisks/underscores aren't surrounded by spaces. This rule attempts to detect where they were surrounded by spaces, but it appears that emphasized text was intended by the author.

**Comment by nzakas** [comment]
Thanks for the suggestion. I feel like this is going a bit beyond what I had in mind for rules in this plugin and would probably be better as a custom rule. Do any existing Markdown linters have a similar rule? If not, then I'd lean towards not including this.

**Comment by lumirlumir** [comment]
I'm 👍 in support of this rule. It's a common syntax mistake, similar to the `no-missing-atx-heading-space` rule.

**Comment by nzakas** [comment]
Okay, for compatibility with Markdownlint I think it's a good idea to add this. 👍

**Comment by SwetaTanwar** [comment]
@Pixel998 Are you still working this, I would like to take this up
